service/src/main/kotlin/fi/espoo/evaka/decision/DecisionService.kt

```diff
@@ -60,13 +59,12 @@ class DecisionService(
         withSpringTx(txManager) {
             val decisionIds = withSpringHandle(dataSource) { h -> finalizeDecisions(h, applicationId) }
 
-            withSpringHandle(dataSource) {
-                asyncJobRunner.plan(it, decisionIds.map { NotifyDecisionCreated(it, user, sendAsMessage) })
+            withSpringHandle(dataSource) { h ->
+                asyncJobRunner.plan(h, decisionIds.map { NotifyDecisionCreated(it, user, sendAsMessage) })
             }
-            runAfterCommit { asyncJobRunner.scheduleImmediateRun() }
 
             decisionIds
-        }
+        }.also { asyncJobRunner.scheduleImmediateRun() }
```

Comment:
Does this guarantee that the transaction has been commited?

Reply:
:facepalm: need more :coffee: